playground/js-sourcemap/vite.config.js

```diff
@@ -12,9 +12,17 @@ export default defineConfig({
     rollupOptions: {
       output: {
         manualChunks(name) {
-          if (name.includes('after-preload-dynamic')) {
+          if (name.endsWith('after-preload-dynamic.js')) {
             return 'after-preload-dynamic'
           }
+          if (name.endsWith('after-preload-dynamic-hashbang.js')) {
+            return 'after-preload-dynamic-hashbang'
+          }
+        },
+        banner(chunk) {
+          if (chunk.name.endsWith('after-preload-dynamic-hashbang')) {
+            return '#!/usr/bin/env node'
+          }
```

Comment:
I used `banner` to inject hashbang and test this case since having `#! ...` inside `after-preload-dynamic-hashbang.js` seems to cause a parse error somewhere in the pipeline. The error looks like this:

```sh
$ pnpm -C playground/js-sourcemap build

> @vitejs/test-js-sourcemap@0.0.0 build /home/hiroshi/code/others/vite/playground/js-sourcemap
> vite build

vite v5.1.3 building for production...
✓ 7 modules transformed.
x Build failed in 56ms
error during build:
RollupError: Expected ident
    at error (file:///home/hiroshi/code/others/vite/node_modules/.pnpm/rollup@4.2.0/node_modules/rollup/dist/es/shared/parseAst.js:337:30)
    at nodeConverters (file:///home/hiroshi/code/others/vite/node_modules/.pnpm/rollup@4.2.0/node_modules/rollup/dist/es/shared/parseAst.js:2072:9)
    at convertNode (file:///home/hiroshi/code/others/vite/node_modules/.pnpm/rollup@4.2.0/node_modules/rollup/dist/es/shared/parseAst.js:957:12)
    at convertProgram (file:///home/hiroshi/code/others/vite/node_modules/.pnpm/rollup@4.2.0/node_modules/rollup/dist/es/shared/parseAst.js:948:48)
    at parseAstAsync (file:///home/hiroshi/code/others/vite/node_modules/.pnpm/rollup@4.2.0/node_modules/rollup/dist/es/shared/parseAst.js:2138:20)
    at Module.tryParseAsync (file:///home/hiroshi/code/others/vite/node_modules/.pnpm/rollup@4.2.0/node_modules/rollup/dist/es/shared/node-entry.js:13357:21)
    at Module.setSource (file:///home/hiroshi/code/others/vite/node_modules/.pnpm/rollup@4.2.0/node_modules/rollup/dist/es/shared/node-entry.js:12953:35)
    at ModuleLoader.addModuleSource (file:///home/hiroshi/code/others/vite/node_modules/.pnpm/rollup@4.2.0/node_modules/rollup/dist/es/shared/node-entry.js:17580:13)
 ELIFECYCLE  Command failed with exit code 1.
```